### ISSUE TITLE
Undocumented deprecation of updateUser?

### ISSUE THREAD

**Issue by lolsheeplol** [intercom/intercom-android#315]
I happened to notice that the `updateUser(Map<String, ?> attributes)` method has been marked deprecated as of version 3.0.19, but I couldn't find any documentation on the repo about this.  Is it recommended to now update users using `updateUser(User user)`?  Are there plans to sunset the deprecated method? 

## Version info
  - Intercom for Android version: 3.0.21
  - Android OS version: N/A
  - Android Device: N/A

## Expected behavior
`updateUser(Map<String, ?> attributes)` with userMap and customAttributes is deprecated

## Actual behavior
`updateUser(User user)` using Intercom User Builder is a new method as of 3.0.19

## Steps to reproduce
 N/A

## Logs

**Comment by lolsheeplol** [comment]
Nevermind, seems my own question was answered in the [3.1.0](https://github.com/intercom/intercom-android/releases/tag/3.1.0) release.

**Comment by DaleCantwell** [comment]
Hi @lolsheeplol 
We have just released 3.1.0 which has the new method updateUser(UserAttributes).

We have deprecated updateUser(User) and updateUser(Map). The reason being the Map was not that clear as to what was and wasn't supported. The updateUser(User) method was leading to confusion with which User model to include as we also have an internal representation of user.

Our docs here show the new usage https://developers.intercom.com/docs/android-configuration#section-update-a-user

Regarding sunsetting I think in a couple releases we will kill off the old methods. Let us know how you find using the new UserAttributes object, any feedback would be much appreciated.

Thanks.
